### PR TITLE
Fix model group index not existing model version query issue and SecureMLRestIT failure ITs

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
The message "Model access control dev" looks not meaningful. Change to a readable message?

File: plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java [hunk 1]
```diff
@@ -66,39 +70,40 @@ public MLSearchHandler(Client client, NamedXContentRegistry xContentRegistry, Mo
...
-                    request.source().query(rewriteQueryBuilder(request.source().query(), null));
-                    client.search(request, listener);
-                }
-            }, e -> {
-                log.error("Fail to search model groups!", e);
-                actionListener.onFailure(e);
-            });
-            try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+            } else if (!clusterService.state().metadata().hasIndex(CommonValue.ML_MODEL_GROUP_INDEX)) {
+                request.source().query(rewriteQueryBuilder(request.source().query(), null));
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java:77]: If no model group index, why need to rewrite query?

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java]: `rewriteQueryBuilder` method is not easy to understand. Add some comments?

Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java]: Make sense, removed the rewriting.

**Review by rbhavna — APPROVED** [review]

**Review by rbhavna — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**